cmds/webboot/network_utils.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+// WriteCounter counts the number of bytes written to it. It implements to the io.Writer
+type WriteCounter struct {
+	Total uint64
+}
+
+func (wc *WriteCounter) Write(p []byte) (int, error) {
+	n := len(p)
+	wc.Total += uint64(n)
+	wc.PrintProgress()
+	return n, nil
+}
+
+// PrintProgress will print how many bytes have been writen to the file
+func (wc WriteCounter) PrintProgress() {
```

Comment:
Why have a separate function for this, does it work to just put these 2 lines directly in the Write function?

Reply:
I think that putting them in a separate function will make the logic clearer, because I think Write() is not neccesarily equal to PrintProgress(). For example, it is possible that we will do other things in Write in addition to the progress bar. Although I can't think of what else needs to do now, I think separating them may be more flexible.